spec/models/post_spec.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+
+# == Schema Information
+#
+# Table name: posts
+#
+#  id         :bigint           not null, primary key
+#  body       :text             not null
+#  status     :integer          default("public"), not null
+#  created_at :datetime         not null
+#  updated_at :datetime         not null
+#  user_id    :bigint
+#
+# Indexes
+#
+#  index_posts_on_user_id  (user_id)
+#
+require 'rails_helper'
+
+RSpec.shared_examples 'Post' do
+  it 'expects post to be valid' do
+    expect(post).to be_valid
+  end
+end
+
+RSpec.describe Post, type: :model do
+  subject(:post) { FactoryBot.create :post }
+
+  context 'when everything is valid' do
+    include_examples 'Post'
+
+    it { is_expected.to have_many(:likes) }
```

Comment:
since the private and public post contexts have the same checks, I guess shared example should be extended with them.

```
shared_examples 'valid post' do
  it { is_expected.to be_valid(:likes) }
  it { is_expected.to have_many(:likes) }
  it { is_expected.to have_many(:pictures) }
  it { is_expected.to belong_to(:user) }
end
```